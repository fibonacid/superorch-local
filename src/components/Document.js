import React, {Component} from 'react';
import {connect} from 'react-redux';
import styled from 'styled-components'
import {addMessage} from "../actions/actions";

const resetAppearence = `
  &:focus,
  &:hover,
  &:active {
    outline:0px !important;
    -webkit-appearance:none;
    box-shadow: none !important;
  }
`;

const StyledContainer = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
`;

const StyledTextArea = styled.textarea`
  flex: 1;
  margin: 0;
  border: none;
  font-family: monospace;
  padding-top: 10px;
  padding-left: 10px;
  ${resetAppearence}
`;

const StyledButton = styled.button`
  display: block;
  -webkit-appearance:none;
  padding: 2.5px;
  background: black;
  color: white;
  font-weight: bold;
  border: 0;
  ${resetAppearence}
  &:active { background: grey; }
`;

class Document extends Component {

  constructor(props) {
    super(props);
    this.state = {
      value: ""
    }
  }

  handleChange(e) {
    e.preventDefault();
    const { document } = this.props;
    this.setState({
      value: document.current + e.target.value
    });
  }

  handleClick(e) {
    const { value } = this.state;
    this.props.send({message: value});
  }

  render() {
    return (
      <StyledContainer>
        <StyledTextArea
          resizable={false}
          value={this.state.value}
          onChange={e => {this.handleChange(e)}} />
        <StyledButton
          onClick={e => {this.handleClick(e)}}
        >SEND</StyledButton>
      </StyledContainer>
    )
  }
}

const mapStateToProps = state => ({
  document: state.chat.document
});

const mapDispatchToProps = dispatch => ({
  send: (message) => dispatch(addMessage(message))
});

export default connect(mapStateToProps, mapDispatchToProps)(Document);
