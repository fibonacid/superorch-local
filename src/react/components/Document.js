import React, {Component} from 'react';
import PropTypes from 'prop-types';
import * as styled from './styles/Document.styles'

const Diff = require('diff');

class Document extends Component {

  constructor(props) {
    super(props);
    this.state = {
      local: props.shared
    };
  }

  componentDidUpdate() {
    const { shared } = this.props;
    const { local } = this.state;
    const diffs = Diff.diffWordsWithSpace(local, shared);
    let text = "";
    diffs.forEach(part => {
      if (!part.removed) {
        text += part.value;
      }
    });
    if (this.state.local !== text) {
      this.setState({
        local: text
      });
    }
  }

  handleChange(event) {
    this.setState({
      local: event.target.value,
    });
  }

  handleClick() {
    const { local } = this.state;
    this.props.send(local);
  }

  render() {
    return (
      <styled.Container
        data-test={'DocumentComponent'}>
        <styled.TextArea
          data-test={'textarea'}
          resizable={false}
          value={this.state.local}
          onChange={this.handleChange.bind(this)} />
        <styled.Button
          data-test={'button'}
          onClick={this.handleClick.bind(this)}
        >SEND</styled.Button>
      </styled.Container>
    )
  }
}

Document.propTypes = {
  shared: PropTypes.string
};

export default Document;
