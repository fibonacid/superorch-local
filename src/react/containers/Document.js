import Document from "../components/Document";
import {addMessage} from "../actions/actions";
import {connect} from "react-redux";

const mapStateToProps = state => ({
  shared: state.chat.document.shared
});

const mapDispatchToProps = dispatch => ({
  send: (message) => dispatch(addMessage(message))
});

export default connect(mapStateToProps, mapDispatchToProps)(Document);
