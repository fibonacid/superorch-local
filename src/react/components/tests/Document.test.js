import React from "react";
import {shallow} from 'enzyme'
import { Provider } from 'react-redux'
import configureStore from "../../store";
import Document from '../Document';

const mockStore = configureStore();

let component;
beforeEach(() => {
  component = shallow(
    <Provider store={mockStore}>
      <Document/>
    </Provider>
  );

});
afterEach(() => {
  component.unmount();
});

describe("Document", () => {

  it('should render', () => {});

  it('should have a textarea and a button', () => {
    /*expect(component.find('styled.TextArea').length).toEqual(1);
    expect(component.find('styled.Button').length).toEqual(1);*/
  }).todo();

  it("should create an entry in component state", () => {
    /*const textarea = component.find('styled.TextArea');
    textarea.simulate('change', { target: textarea, value: 'foo' });
    expect(component.state.local).toEqual('foo');*/
  }).todo();
});
