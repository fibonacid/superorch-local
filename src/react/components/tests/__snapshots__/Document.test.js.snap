// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component has props Should render without errors 1`] = `
<Documentstyles__Container
  data-test="DocumentComponent"
>
  <Documentstyles__TextArea
    data-test="textarea"
    onChange={[Function]}
    resizable={false}
  />
  <Documentstyles__Button
    data-test="button"
    onClick={[Function]}
  >
    SEND
  </Documentstyles__Button>
</Documentstyles__Container>
`;
